trio/socket.py

```diff
@@ -9,6 +9,62 @@
 from . import _socket
 import sys as _sys
 
+# expose all uppercase names from standardlib socket to trio.socket
+import socket as _stdlib_socket
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = True  # Not precise but only for python 3.5.0
```

Comment:
We should actually set it to `False`! When we're running on the python interpreter, we don't want to execute the code inside the `if` block. And the static tools like mypy have hard-coded logic that when they see this variable, they act like it's `True` regardless of what it's actually set to.

Reply:
Yes you are right, but 3.5.0 does not have that variable. Either we find a workaround or we have to drop 3.5.0 compatibility as https://github.com/python/mypy/issues/4714 did.